massa-final-state/src/final_state.rs

```diff
@@ -25,6 +27,10 @@ use tracing::{debug, info, warn};
 use massa_models::config::PERIODS_BETWEEN_BACKUPS;
 use massa_models::timeslots::get_block_slot_timestamp;
 
+// TODO when creating a final ledger at genesis, initialize the execution hash trail to Hash::zero()
```

Comment:
We should use a tool to generate an initial DB on disk and just let the system load it from there, like it would for a restart to avoid having this weird special case. Can we open a followup ?

Reply:
> The easiest to integrate with your current code is IMO to do it directly in the load_initial_ledger() method in ledger_db.rs : just add the key/value to the batch there.

Until we do the followup I mentioned, is there a way to do it without contaminating either low level DB or the unrelated Ledger module ?

Reply:
You could call another function here in massa-bootstrap>src>client.rs, that handles this, but it would have to be in final_state or in the db.
```rust
if !bootstrap_config.keep_ledger {
    // load ledger from initial ledger file
    final_state_guard
        .ledger
        .load_initial_ledger()
        .map_err(|err| {
            BootstrapError::GeneralError(format!(
            "could not load initial ledger: {}",
            err
        ))
    })?;
}
```

Reply:
> We should use a tool to generate an initial DB on disk and just let the system load it from there, like it would for a restart to avoid having this weird special case. Can we open a followup ?

https://github.com/massalabs/massa/issues/4237

Reply:
> You could call another function here in massa-bootstrap>src>client.rs, that handles this, but it would have to be in final_state or in the db.
> 
> ```rust
> if !bootstrap_config.keep_ledger {
>     // load ledger from initial ledger file
>     final_state_guard
>         .ledger
>         .load_initial_ledger()
>         .map_err(|err| {
>             BootstrapError::GeneralError(format!(
>             "could not load initial ledger: {}",
>             err
>         ))
>     })?;
> }
> ```

Thanks for the feedback. Checking it out.

I guess that keep_ledger should be renamed keep_state (?)

A general observation: without having worked on it previously, it is essentially impossible to add something to the state without making many more or less silent mistakes. I hope that the refactorings will help make it less error-prone :)

Reply:
Sure, keep_ledger could probably become keep_state or keep_db.

Aside from the refactor, I think we could improve the documentation too.